cxx/pclean/pclean_lib_test.cc

```diff
@@ -166,3 +166,61 @@ observe
   BOOST_TEST(samples.data["City"].size() == 10);
   BOOST_TEST(samples.data["State"].size() == 10);
 }
+
+BOOST_AUTO_TEST_CASE(test_make_pclean_samples) {
+  std::mt19937 prng;
+
+  std::stringstream ss(R"""(
+class School
+  name ~ string
+  degree_dist ~ categorical(k=100)
+
+class Physician
+  school ~ School
+  degree ~ stringcat(strings="MD PT NP DO PHD")
+  specialty ~ stringcat(strings="Family Med:Internal Med:Physical Therapy", delim=":")
+  # observed_degree ~ maybe_swap(degree)
+
+class City
+  name ~ string
+  state ~ stringcat(strings="AL AK AZ AR CA CO CT DE DC FL GA HI ID IL IN IA KS KY LA ME MD MA MI MN MS MO MT NE NV NH NJ NM NY NC ND OH OK OR PA RI SC SD TN TX UT VT VA WA WV WI WY")
+
+class Practice
+  city ~ City
+
+class Record
+  physician ~ Physician
+  location ~ Practice
+
+observe
+  physician.specialty as Specialty
+  physician.school.name as School
+  physician.degree as Degree
+  location.city.name as City
+  location.city.state as State
+  from Record
+)""");
+
+  PCleanSchema pclean_schema;
+  BOOST_TEST(read_schema(ss, &pclean_schema));
+
+  GenDB gendb(&prng, pclean_schema);
+
+  T_encoding enc = make_dummy_encoding_from_gendb(gendb);
+
+  BOOST_TEST(enc.second.size() == 0);
+
+  std::map<std::string, ObservationVariant> obs = {
+    {"Specialty", "Internal Med"},
+    {"School", "Harvard"},
+    {"Degree", "MD"},
+    {"City", "Cambridge"},
+    {"State", "MA"}};
+
+  gendb.incorporate(prng, {0, obs}, true);
+
+  BOOST_TEST(enc.second["School"][0] == "School:0");
```

Comment:
Consider looping over `domain_crps.at("School").tables` to make sure all the values are accounted for.